src/app/shared/model/feeder-podcast.model.ts

```diff
@@ -0,0 +1,119 @@
+import { Observable} from 'rxjs';
+import { HalDoc } from '../../core';
+import { BaseModel } from './base.model';
+import { REQUIRED, URL, TOKENY } from './invalid';
+
+export class FeederPodcastModel extends BaseModel {
+
+  // read-only
+  id: number;
+  previewUrl: string;
+  publishedUrl: string;
+
+  // writeable
+  SETABLE = ['category', 'subCategory', 'explicit', 'path', 'link', 'newFeedUrl'];
+  category: string = '';
+  subCategory: string = '';
+  explicit: string = '';
+  path: string = '';
+  link: string = '';
+  newFeedUrl: string = '';
+
+  VALIDATORS = {
+    path: [TOKENY('Use letters, numbers and underscores only')],
+    link: [REQUIRED(), URL('Not a valid URL')],
+    newFeedUrl: [URL('Not a valid URL')]
+  };
+
+  constructor(private series: HalDoc, distrib: HalDoc, podcast?: HalDoc, loadRelated = true) {
+    super();
+    this.init(distrib, podcast, loadRelated);
+  }
+
+  key() {
+    if (this.doc) {
+      return `prx.podcast.${this.doc.id}`;
+    } else if (this.series) {
+      return `prx.podcast.new.${this.series.id}`; // new in series
+    } else {
+      return null; // Cannot create podcast until parent series exists
+    }
+  }
+
+  related() {
+    return {};
+  }
+
+  decode() {
+    this.id = this.doc['id'];
+    this.previewUrl = (this.doc.expand('self') || '').replace(/api\/v1\//, '');
+    this.publishedUrl = this.doc['publishedUrl'];
+    this.explicit = this.doc['explicit'] || '';
+    if (this.explicit) {
+      this.explicit = this.explicit.charAt(0).toUpperCase() + this.explicit.slice(1);
+    }
+    this.link = this.doc['link'] || '';
+    this.newFeedUrl = this.doc['newFeedUrl'] || '';
+
+    // pretend path was blank if it was just the podcast id
+    this.path = this.doc['path'] || '';
+    if (`${this.path}` === `${this.id}`) {
+      this.path = '';
+    }
+
+    // just ignore all but first category/subcategory
+    let cat = (this.doc['itunesCategories'] || [])[0];
+    if (cat) {
+      this.category = cat['name'] || '';
+      if (cat['subcategories']) {
+        this.subCategory = cat['subcategories'][0] || '';
+      } else {
+        this.subCategory = '';
+      }
+    } else {
+      this.category = '';
+      this.subCategory = '';
+    }
+  }
+
+  encode(): {} {
+    let data = <any> {};
+
+    // unset things with nulls instead of blank strings
+    data.explicit = this.explicit || null;
+    if (data.explicit) {
+      data.explicit = data.explicit.toLowerCase();
+    }
+    data.link = this.link || null;
+    this.newFeedUrl = this.newFeedUrl || null;
```

Comment:
typo! should be `data.newFeedUrl = this.newFeedUrl || null`, no? (I was wondering why my http://puppies.com wasn't persisting!)